python/foglamp/services/core/api/certificate_store.py

```diff
@@ -89,3 +90,30 @@
     # and reboot
     return web.json_response({"result": "{} and {} have been uploaded successfully"
                              .format(key_filename, cert_filename)})
+
+
+async def delete_certificate(request):
+    """ Delete a certificate
+
+    :Example:
+          curl -X DELETE http://localhost:8081/foglamp/certificate/foglamp.cert
+    """
+    cert_name = request.match_info.get('cert_name', None)
+
+    # accepted extensions are '.key and .cert'
+    valid_extensions = ('.key', '.cert')
+    if not cert_name.endswith(valid_extensions):
+        raise web.HTTPBadRequest(reason="Accepted file extensions are .key and .cert")
+
+    if _FOGLAMP_DATA:
+        certs_dir = os.path.expanduser(_FOGLAMP_DATA + '/etc/certs')
+    else:
+        certs_dir = os.path.expanduser(_FOGLAMP_ROOT + '/data/etc/certs')
+
+    cert_path = certs_dir + '/{}'.format(cert_name)
+
+    if os.path.isfile(cert_path):
+        os.remove(cert_path)
+        return web.json_response({'message': "'{}' certificate is deleted successfully".format(cert_name)})
+    else:
+        raise web.HTTPNotFound(reason='{} does not exist'.format(cert_name))
```

Comment:
We don't do any checks to see of a certificate is in use. The consequences of deleting the certificate used for the admin API really need to be understood. I suspect it will result in an unusable API, hence we probably need to guard against this.